core/src/ru/spbau/blackout/abilities/simpleshell/SimpleShellAbility.java

```diff
@@ -0,0 +1,66 @@
+package ru.spbau.blackout.abilities.simpleshell;
+
+import com.badlogic.gdx.math.Vector2;
+
+import org.jetbrains.annotations.Contract;
+
+import ru.spbau.blackout.abilities.Ability;
+import ru.spbau.blackout.abilities.SimpleInstantAbility;
+import ru.spbau.blackout.abilities.DynamicAbilityObject;
+import ru.spbau.blackout.entities.Character;
+import ru.spbau.blackout.entities.GameObject;
+
+public abstract class SimpleShellAbility extends SimpleInstantAbility {
+
+    protected SimpleShellAbility(Definition def, Character character) {
+        super(def, character);
+    }
+
+    /**
+     * Creates shell with start speed.
+     */
+    @Override
+    public void cast(Vector2 targetOffset) {
+        super.cast(targetOffset);
+
+        Vector2 unitPosition = getCharacter().getPosition();
+        targetOffset.add(unitPosition);  // real target
+        GameObject.Definition shellDef = ((Definition) getDef()).shellDef;
+        DynamicAbilityObject shell = (DynamicAbilityObject) shellDef.makeInstanceWithNextUid(targetOffset);
+
+        float startSpeed = ((Definition) getDef()).startSpeed();
+        Vector2 startVelocity = targetOffset.sub(unitPosition).scl(startSpeed / targetOffset.len());
+        shell.velocity.add(startVelocity);
+    }
+
+    public abstract static class Definition extends Ability.Definition {
+
+        private static final long serialVersionUID = 1000000000L;
```

Comment:
I'm not sure, but it seems like we have some problems with automatically generated version UIDs due to the different environment on the server and devices.